packages/nimble-components/src/table/specs/table-keyboard-navigation-hld.md

```diff
@@ -0,0 +1,180 @@
+
+When a cell is focused:
+
+-   If the focus is on the first cell, pressing `ArrowLeft` will focus the entire row
+-   If the focus is on any cell other than the first cell, pressing `ArrowLeft` will move focus one cell to the left
+-   If the focus is on any cell other than the last, pressing `ArrowRight` will move focus to the next cell to the right
+-   If the focus is on the last cell, pressing `ArrowRight` does nothing
+-   Note: If a focused cell has multiple interactive elements for its contents, those elements will _not_ receive focus from pressing `ArrowLeft` or `ArrowRight`, and instead focus will be shifted to the appropriate neighboring cell (or do nothing if focus is at the extents of the row).
+
+Special case of focusing a cell:
+
+-   If a cell does not have an action menu, and only contains a single interactive/focusable element of the following types: button (and variants), switch, checkbox, or anchor/link: that child element will be focused _instead_ of the cell.
+
+To interact with a cell's interactive contents, starting from the state of a cell being focused:
+
+-   If the cell has an interactive element (including the action menu button):
+    -   Pressing `Enter` will focus the 1st/ primary interactive element, _and_ trigger its functionality (e.g. button press, link navigation)
+    -   Pressing `F2` will focus the 1st/ primary interactive element, but not trigger its functionality
+    -   At this point, the table is no longer in 'navigation mode' (will not handle arrow key presses). `Tab`/`Shift-Tab` will also be handled by the active element by default, but may bubble up and be handled by the table (resulting in a focus change) if the content does not process `Tab`/`Shift-Tab` presses. To return to 'navigation mode', and focus the cell again:
+        -   `Esc` is pressed: If the user had edited content (e.g. textbox), it may be reverted.
+        -   `Enter` is pressed: If the user was editing content or interacting with a control, those changes will be committed.
+-   For all cells with action menu buttons:
+    -   `Ctrl-Enter` will trigger the action menu button and open the action menu. Note: If the action menu button is the only interactive element in the cell, `Enter` will do the same thing.
+-   Pressing `Tab` will focus the 1st/ primary interactive element in the cell. Subsequent `Tab` presses will focus subsequent interactive elements in the cell, including the action menu button. `Shift-Tab` will focus the previous interactive element. Note: `Tab` will not cycle within the cell (it will go to the next focusable elements in the row, or focus will leave the table, as outlined in the "`Tab`/`Shift-Tab` behavior" section above).
+
+If a cell is focused and `ArrowDown` or `ArrowUp` is pressed, when there is an available row in the navigation direction, focus will move to the cell in the current focused column of the new row.
```

Comment:
Related to [Molly's comment in the open issues section](https://github.com/ni/nimble/pull/1960/files#r1541750707) we should explicitly mention the plan for scrolling, both vertically and horizontally, if the cell the user is moving to is not currently visible.

Reply:
Added a section about scrolling the table (generally anytime we focus/ refocus a row/cell, we'll scroll to it if needed).

I've mostly been considering vertical scrolling, not horizontal. I'll keep this conversation open for now to check if any further clarification is needed for horizontal scrolling.

Reply:
Resolving because I think the current wording covers horizontal scrolling too. Vertical scrolling will be via virtualizer APIs, horizontal scrolling will be via something like `Element.scrollIntoView()`.